### PR TITLE
DUOS-1481[risk=no] Added DAR Drafts table to NewResearcherConsole

### PR DESCRIPTION
**Pull request by JVThomas**
Addresses [DUOS-1481](https://broadworkbench.atlassian.net/browse/DUOS-1481)

PR adds the DAR Drafts table to the updated Researcher Console with redirect and delete functions as seen in the current console. PR also included directory updates to better organize component (NewResearcherConsole) within the file structure.

Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/dar_collection_table/DarCollectionTable.js [hunk 1]
```diff
@@ -153,7 +153,6 @@ export default function DarCollectionTable(props) {
   const [tableSize, setTableSize] = useState(10);
   const [showConfirmation, setShowConfirmation] = useState(false);
   const [selectedCollection, setSelectedCollection] = useState({});
-  // const searchRef = useRef(''); //May not need this, could just pass in the value from the parent
```
Comment (rushtong) [inline src/components/dar_collection_table/DarCollectionTable.js:156]: nit: do we need this comment?

File: src/components/dar_drafts_table/DarDraftTable.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    }
+  };
+
+  //delete function that fires on modal confirmation, removes draft from listing via prop delete function
+  const deleteOnClick = async() => {
+    const {id, data, referenceId} = selectedDraft;
+    const identifier = getIdentifier({id: id || referenceId, data});
+    await deleteDraft({ referenceId, identifier, });
+    setShowConfirmation(false);
+  };
```
Comment (rushtong) [inline src/components/dar_drafts_table/DarDraftTable.js:227]: I don't think this isn't refreshing the underlying list of draft dars ... after delete, the deleted draft remains visible onscreen.

**Review by rushtong — COMMENTED** [review]
Looks good! One functionality fix and one minor nit, but otherwise 👍🏽

File: src/components/dar_collection_table/DarCollectionTable.js (hunk 1 above)
Comment (JVThomas) [reply src/components/dar_collection_table/DarCollectionTable.js:156]: We don't, which is why it's removed

File: src/components/dar_drafts_table/DarDraftTable.js (hunk 2 above)
Comment (JVThomas) [reply src/components/dar_drafts_table/DarDraftTable.js:227]: Yeah, that's a bug, should be fixed with the latest commit
Kapture 2021-11-09 at 07 40 35

**Review by rushtong — APPROVED** [review]
Thanks for the fixes! 👍🏽

**Pull request merged**